Library/Homebrew/dev-cmd/bottle.rb

```diff
@@ -312,7 +312,7 @@ def bottle_formula(f, args:)
     else
       ohai "Determining #{f.full_name} bottle rebuild..."
       FormulaVersions.new(f).formula_at_revision("origin/HEAD") do |upstream_f|
-        if f.pkg_version == upstream_f.pkg_version
+        if f.pkg_version == upstream_f.pkg_version && !upstream_f.bottle_unneeded?
```

Comment:
Should we add `TODO` here that says this can be removed when we no longer have `bottle :unneeded` in Homebrew/core?

Or do we want to keep this indefinitely in case other taps might someday decide to remove their `bottle :unneeded`s too?

In either case, this might warrant a comment.

Reply:
I think it needs to stay for third-party taps. I don't really feel that a comment is necessary because it's clear what the intent is without needing any context for this issue (if the old formula was `bottle :unneeded` and the new one isn't, the rebuild should be zero.

I'm not opposed to a comment if you'd like, though, just not sure what it would say.

Reply:
Hmmm, yea, you're right. I guess this is clear enough.